ipatests/test_integration/test_adtrust_install.py

```diff
@@ -314,3 +314,376 @@ def test_ipa_user_s4u2self_pac(self):
+        result = self.master.run_command(
+            ["ipa-adtrust-install", "-a",
+             self.master.config.admin_password, "-U"], raiseonerr=False)
+        assert msg in result.stderr_text
+
+    def test_samba_credential_cache_is_removed_post_uninstall(self):
+        """
+        Test checks that samba credential cache is removed after
+        ipa-server is uninstalled.
+        https://pagure.io/freeipa/issue/3479
+        """
+        result = tasks.install_master(self.master, setup_dns=True)
+        assert result.returncode == 0
+        self.master.run_command(
+            ["ipa-adtrust-install", "-a",
+             self.master.config.admin_password, "-U"]
+        )
+        assert self.master.transport.file_exists(paths.KRB5CC_SAMBA)
+        tasks.uninstall_replica(self.master, self.replicas[0])
+        tasks.uninstall_master(self.master)
+        assert not self.master.transport.file_exists(paths.KRB5CC_SAMBA)
+
+    def test_adtrust_install_without_integrated_dns(self):
+        """
+        Test checks ipa-adtrust-install displays the necessary
+        service records to be added on a IPA server
+        without integrated dns setup.
+        """
+        realm = self.master.domain.realm.lower()
+        hostname = self.master.hostname
+        msg = (
+            "Done configuring CIFS.\n"
+            "DNS management was not enabled at install time.\n"
+            "Add the following service records to your DNS server "
+            "for DNS zone {0}: \n"
+            "_ldap._tcp.Default-First-Site-Name._sites.dc._msdcs.{0}. "
+            "3600 IN SRV 0 100 389 {1}.\n"
+            "_ldap._tcp.dc._msdcs.{0}. 3600 IN SRV 0 100 389 {1}.\n"
+            "_kerberos._tcp.Default-First-Site-Name._sites.dc._msdcs.{0}. "
+            "3600 IN SRV 0 100 88 {1}.\n"
+            "_kerberos._udp.Default-First-Site-Name._sites.dc._msdcs.{0}. "
+            "3600 IN SRV 0 100 88 {1}.\n"
+            "_kerberos._tcp.dc._msdcs.{0}. 3600 IN SRV 0 100 88 {1}.\n"
+            "_kerberos._udp.dc._msdcs.{0}. 3600 IN SRV 0 100 88 {1}.\n\n"
+            "================================================================"
+            "=============\n"
+            "Setup complete\n\n"
+        ).format(realm, hostname)
+        tasks.uninstall_master(self.master)
+        result = tasks.install_master(self.master, setup_dns=False)
+        assert result.returncode == 0
+        cmd = self.master.run_command(
+            ["ipa-adtrust-install", "-a",
+             self.master.config.admin_password, "-U"]
+        )
+        assert msg in cmd.stdout_text
+
+    def test_adtrust_install_with_debug_option(self):
+        """
+        Test checks that ipa-adtrust-install runs with debug option
+        without any error.
+        """
+        result = self.master.run_command(
+            ["ipa-adtrust-install", "-a",
+             self.master.config.admin_password, "-U", "-d"]
+        )
+        assert result.returncode == 0
```

Comment:
You can remove this line, if run_command is called without raise_error=False

Reply:
assert statement has now been removed for test_adtrust_install_with_debug_option testcase, since raiseonerr=False